node/src/components/consensus/highway_core/highway.rs

```diff
@@ -904,4 +905,23 @@ pub(crate) mod tests {
             validate(&wunit0, &CAROL_SEC, &wunit1, &CAROL_SEC)
         );
     }
+
+    #[test]
+    fn invalid_ping_ndrs1077_regression() {
+        let now: Timestamp = 500.into();
+
+        let state: State<TestContext> = State::new_test(WEIGHTS, 0);
+        let highway = Highway {
+            instance_id: TEST_INSTANCE_ID,
+            validators: test_validators(),
+            state,
+            active_validator: None,
+        };
+
+        // Ping by validator that is not bonded, with an index that is outside of boundaries of the
+        // state.
+        let ping: Vertex<TestContext> = Vertex::Ping(Ping::new(DAN, now, &DAN_SEC));
```

Comment:
Maybe just so this test remains relevant even if we change the default weights:
```rust
        assert!(DAN_SEC.0 >= WEIGHTS.len());
```

Reply:
I was just adding this right now :)

Reply:
Added in [`37c07c1` (#1240)](https://github.com/CasperLabs/casper-node/pull/1240/commits/37c07c15d45fb29a9f0d251252722e59b0f1411f)